pkg/reconciler/status.go

```diff
@@ -129,8 +129,15 @@ func (s *Status) SetDeleteCompleted(err error) {
 	} else {
 		// assume resource will be deleted, hence nothing to update
 	}
+}
 
-	s.UpdateFunc(s.S)
+// SetPackageYanked appends a new condition to indicate a package is yanked
+func (s *Status) SetPackageYanked(meta metav1.ObjectMeta, reason string) {
```

Comment:
feels a bit weird to have a "yanked" knowledge within generic pkg/reconciler package. wdyt?